src/components/modals/Dialog.jsx

```diff
@@ -0,0 +1,120 @@
+      props.onDismiss();
+    }
+  };
+
+  useLayoutEffect(() => {
+    // This toggles scrolling on and off based on whether the modal
+    // is shown or not
+    if (props.isOpen) noScroll.on();
+    noScroll.off();
+  }, [props.isOpen]);
+
+  return ReactDOM.createPortal(
+    <CSSTransition timeout={200} in={props.isOpen} classNames="dialog" unmountOnExit>
+      <React.Fragment>
+        <FocusTrap>
+          <div>
+            <div className="dialog-overlay" onClick={handleClick}></div>
+            <div
+              className="dialog elevation--3"
+              role={props.role}
+              aria-labelledby={props.title && 'a11y-dialog-title'}
+              aria-describedby="a11y-dialog-desc"
+              tabIndex="-1"
+              aria-modal="true"
+              onKeyDown={handleKeyDown}
+            >
+              {(props.title || props.canDismiss) && (
+                <React.Fragment>
+                  <div className="dialog-header">
+                    <Section border="bottom">
+                      <Flex justify="spaceBetween" align="start">
+                        <FlexItem>
+                          {props.title && (
+                            <div
+                              className="dialog-title type--head4"
+                              id="a11y-dialog-title"
+                            >
+                              {props.title}
+                            </div>
+                          )}
+                        </FlexItem>
+                        {props.canDismiss && (
+                          <FlexItem shrink>
+                            <button className="dialog-icon" onClick={handleClick}>
+                              <DenyIcon size="xs" />
+                            </button>
+                          </FlexItem>
+                        )}
+                      </Flex>
+                    </Section>
+                  </div>
+                </React.Fragment>
+              )}
+              <div className="dialog-content">
+                <Section>{props.content}</Section>
+              </div>
+              {props.footerContent && (
+                <div className="dialog-footer">
+                  <Section border="top" bgColor="lightGray">
+                    {props.footerContent}
+                  </Section>
+                </div>
+              )}
+            </div>
+          </div>
+        </FocusTrap>
+      </React.Fragment>
+    </CSSTransition>,
+    // eslint-disable-next-line no-undef
+    document.body
+  );
+};
+
+Dialog.defaultProps = {
+  role: 'dialog',
+};
+
+Dialog.propTypes = {
+  /** Controls the role of the modal */
+  role: PropTypes.oneOf(['dialog', 'alertdialog']),
+  /** Controls whether the modal (and overlay) are shown or not */
+  isOpen: PropTypes.bool,
+  /** Controls whether the modal can be cleared passively (esc key, clicking
+   * backdrop, showing x on modal, clicking on modal x to close) */
+  canDismiss: PropTypes.bool,
```

Comment:
Instead of having `props.canDismiss`, would it be better if it's calculated based on if `props onDismiss` is passed a function?

Reply:
`onDismiss` is an optional callback - our thinking was that an engineer shouldn't have to pass a noop to trigger a dismissible dialog

Reply:
Yeah good question. The only thing that would determine if we can do this, is if the consumer fully controls the dismiss otherwise... so on modals like prompt... let's just imagine FDS controlled the buttons for dismissing... even though we don't let the user click the x mark, we'd still want the ability to dismiss the modal.

That being said... I'm not "sure" that's a thing. You may be right that we can derive this just from the function.

Reply:
If `props.isOpen` determines if a modal is open and that is being controlled outside of this component, to dismiss this, they will need to pass a `props.onDismiss` to toggle `props.isOpen`. If not, I suppose `handleClick` and `handleKeyDown` will need to check if `props.onDismiss` is passed a function?

Reply:
😄 this is difficult to explain.

I think the best course of action is to move forward with your solution, and we can re-examine this at another time. What I'm thinking of isn't needed now and could in fact be a pre-optimization.

(and yeah handleClick & handleKeyDown will need to check onDismiss, which is fine).